File: sandbox.register/test_src/kz/greetgo/sandbox/register/test/util/TestNgRecreateDbListener.java
///MODIFY replace sandbox {PROJECT_NAME}
package kz.greetgo.sandbox.register.test.util;

///MODIFY replace sandbox {PROJECT_NAME}
import kz.greetgo.sandbox.register.test.beans.develop.DbWorker;
import org.testng.ISuite;
import org.testng.ISuiteListener;

public class TestNgRecreateDbListener implements ISuiteListener {
  @Override
  public void onFinish(ISuite suite) {}

  @Override
  public void onStart(ISuite suite) {
    if (!"All Db Test Suite".equals(suite.getName())) {
      return;
    }

    try {
      prepareDatabases();
    } catch (RuntimeException e) {
      throw e;
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }

  private static void prepareDatabases() throws Exception {
    {
      DbWorker postgresDbWorker = TestsBeanContainerCreator.create().dbWorker();
      postgresDbWorker.cleanConfigsForTeamcity();
      postgresDbWorker.recreateAll();
    }
  }
}

File: sandbox.register/test_src/kz/greetgo/sandbox/register/test/util/TestNgRecreateDbListener.kt
///MODIFY replace sandbox {PROJECT_NAME}
package kz.greetgo.sandbox.register.test.util

import org.testng.ISuite
import org.testng.ISuiteListener

class TestNgRecreateDbListener : ISuiteListener {
  override fun onFinish(suite: ISuite) {}

  override fun onStart(suite: ISuite) {
    if (suite.name != "All Db Test Suite") {
      return
    }

    prepareDatabases()

  }

  private fun prepareDatabases() {
    val postgresDbWorker = TestsBeanContainerCreator.create().dbWorker()
    postgresDbWorker.cleanConfigsForTeamcity()
    postgresDbWorker.recreateAll()
  }
}
